### PR TITLE
Fix webflux error handling

### PR DESCRIPTION
**Pull request by nsev**
Currently (at least) `NflowNotFoundException`s seem to get mapped to 500 instead of 404.

### CODE REVIEW

File: nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java [hunk 1]
```diff
@@ -183,15 +183,22 @@ public ListWorkflowInstanceResponse fetchWorkflowInstance(final long id, final S
     return listWorkflowConverter.convert(instance, includes);
   }
 
+  protected int resolveExceptionHttpStatus(Throwable t) {
+    if (t instanceof IllegalArgumentException) {
+      return 400;
+    } else if(t instanceof NflowNotFoundException) {
```
Comment (efonsell) [inline nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java:189]:
```suggestion
    } else if (t instanceof NflowNotFoundException) {
```

File: nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java [hunk 2]
```diff
@@ -183,15 +183,22 @@ public ListWorkflowInstanceResponse fetchWorkflowInstance(final long id, final S
...
+      return 404;
+    }
+
```
Comment (efonsell) [inline nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java:192]:
```suggestion
```

File: nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java [hunk 3]
```diff
@@ -183,15 +183,22 @@ public ListWorkflowInstanceResponse fetchWorkflowInstance(final long id, final S
     return listWorkflowConverter.convert(instance, includes);
   }
 
+  protected int resolveExceptionHttpStatus(Throwable t) {
+    if (t instanceof IllegalArgumentException) {
+      return 400;
```
Comment (efonsell) [inline nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java:188]: Not added in this PR, but prefer using some existing constants for HTTP status codes. Same for 404 and 500.

**Review by efonsell — CHANGES_REQUESTED** [review]

File: nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java (hunk 3 above)
Comment (nsev) [reply nflow-rest-api-common/src/main/java/io/nflow/rest/v1/ResourceBase.java:188]: Sure - fixing.

**Comment by efonsell** [comment]
Would it be possible to add some tests for this? I think the tests for netty are lacking in general, so not mandatory in this PR. But would be nice to have at least something...

**Review by efonsell — APPROVED** [review]

**Pull request merged**